components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/cdc/CDCProcessor.java

```diff
@@ -0,0 +1,274 @@
+    private static final Log LOGGER = LogFactory.getLog(CDCProcessor.class);
+
+    private enum operations {create, update, delete, truncate};
+    private enum opCodes {c, u, d, t};
+
+    public CDCProcessor(InboundProcessorParams params) {
+        this.name = params.getName();
+        this.injectingSeq = params.getInjectingSeq();
+        this.onErrorSeq = params.getOnErrorSeq();
+        this.synapseEnvironment = params.getSynapseEnvironment();
+        this.cdcProperties = params.getProperties();
+        setProperties();
+        try {
+            this.interval = Long.parseLong(cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_INTERVAL));
+        } catch (NumberFormatException nfe) {
+            throw new SynapseException("Invalid numeric value for interval.", nfe);
+        }
+        this.sequential = true;
+        if (cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL) != null) {
+            this.sequential = Boolean
+                    .parseBoolean(cdcProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL));
+        }
+        this.coordination = true;
+        if (cdcProperties.getProperty(PollingConstants.INBOUND_COORDINATION) != null) {
+            this.coordination = Boolean.parseBoolean(cdcProperties.getProperty(PollingConstants.INBOUND_COORDINATION));
+        }
+    }
+
+    private void setProperties () {
+        LOGGER.info("Initializing the properties");
+        try {
+            if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_OFFSET_STORAGE, FILE_OFFSET_STORAGE_CLASS);
+            }
+            if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE).equals(FILE_OFFSET_STORAGE_CLASS)) {
+                String filePath;
+                if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME) == null) {
+                    filePath = "cdc/offsetStorage/" + this.name + "_.dat";
+                } else {
+                    filePath = this.cdcProperties.getProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME);
+                }
+                createFile(filePath);
+                this.cdcProperties.setProperty(DEBEZIUM_OFFSET_STORAGE_FILE_FILENAME, filePath);
+            }
+
+            if (this.cdcProperties.getProperty(DEBEZIUM_OFFSET_FLUSH_INTERVAL_MS) == null) {
+                this.cdcProperties.setProperty(DEBEZIUM_OFFSET_FLUSH_INTERVAL_MS, "1000");
+            }
+
+            String passwordString = this.cdcProperties.getProperty(DEBEZIUM_DATABASE_PASSWORD);
+            SynapseEnvironment synapseEnvironment = this.synapseEnvironment;
+            MessageContext messageContext = synapseEnvironment.createMessageContext();
+
+            this.cdcProperties.setProperty(DEBEZIUM_DATABASE_PASSWORD, resolveSecureVault(messageContext, passwordString));
```

Comment:
Is there any reason not use the [SecureVaultResolver](https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/util/resolver/SecureVaultResolver.java) class to resolve these?

Reply:
checked it. it requires the modification to the existing format, (adding curly brackets to the entire string), which requires as follows in synapse config file : 
```  <parameter name="database.password">{wso2:vault-lookup('mysql_password')}</parameter>```

the existing config is as follows : 
```  <parameter name="database.password">wso2:vault-lookup('mysql_password')</parameter>```

Shall i go ahead with the SecureVaultResolver. what do you recommend ?

Reply:
Yes, better to go with the SecureVaultResolver. When we need to have a expression or vault-lookup we usually define it with **{ }**.

Reply:
thanks, fixed